### PR TITLE
Add simple mergify config. It uses branch protection rules from github configuration.

### PR DESCRIPTION
**Pull request by vivekvpandya**

### CODE REVIEW

**Comment by vivekvpandya** [comment]
We need to update branch protection rules as per our requirements and mergify will use them to automatically merge PRs.
https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/defining-the-mergeability-of-pull-requests/managing-a-branch-protection-rule

**Comment by vivekvpandya** [comment]
This should fix https://github.com/zeitgeistpm/zeitgeist/issues/688 , in github branch protection rules we should choose setting to enforce checking latest test checks success.

**Comment by sea212** [comment]
Started a review. You are right, it should resolve #688 if configured properly. As stated in the [mergify documentation](https://twemoji.maxcdn.com/v/13.1.0/svg/1f517.svg), mergify will always merge main into a PR before giving green lights for a merge:

File: .mergify.yml
Comment (sea212) [inline .mergify.yml:6]: AFAIK it is not an automatic merge, it just puts the PR in the merge queue once `conditions` are met.

```suggestion
  - name: Use merge queue
```

File: .mergify.yml
Comment (sea212) [inline .mergify.yml:8]: We can expand this further.

```suggestion
    conditions:
      - base=main
      - label=s:accepted
```

File: .mergify.yml
Comment (sea212) [inline .mergify.yml:11]: I every once in a while saw invalid labels in merged pull requests. We can also remove some labels as a guard:
```suggestion
    actions:
      label:
        remove:
          - s:in-progress
          - s:review-needed
      queue:
        name: default
```

**Review by sea212 — COMMENTED** [review]
Good find Vivek, I like the solution much!

**Comment by vivekvpandya** [comment]
> Looks cool. Agree with @sea212's suggestions. While we're at it, can we add rules for the following?
> 
>     * Ensure that merge is squashed (wouldn't surprise me if this is impossible)
> 
>     * Ensure that the PR was reviewed

both of these rules are available in github's branch protection mechanism

File: .mergify.yml
Comment (vivekvpandya) [reply .mergify.yml:8]: I think this lable check can be done in github branch protection rules?

File: .mergify.yml
Comment (vivekvpandya) [reply .mergify.yml:11]: can't that be done by github rules?

**Comment by sea212** [comment]
@maltekliemann Squash and merge is already enforced, I requested setting a minimum review count to the admin.

File: .mergify.yml
Comment (sea212) [reply .mergify.yml:11]: Not that I know. If you can reference documentation about it I prefer to skip it, because I am in favor of setting all possible rules in the branch protection rules.

File: .mergify.yml
Comment (sea212) [reply .mergify.yml:8]: Not that I know. If you can reference documentation about it I prefer to skip it, because I am in favor of setting all possible rules in the branch protection rules.

**Comment by vivekvpandya** [comment]
https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/defining-the-mergeability-of-pull-requests/managing-a-branch-protection-rule
here rule 11 require merge queue will be required right @sea212 ?

**Comment by sea212** [comment]
Passing status checks and minimum review count have been configured in the repository.
Merge queue is in public beta, we have requested access and are await response.

**Comment by vivekvpandya** [comment]
> Passing status checks and minimum review count have been configured in the repository. Merge queue is in public beta, we have requested access and are await response.

how do we check the status for this?

**Comment by sea212** [comment]
Afaik we have to ask the GitHub admin (Logan).

**Comment by vivekvpandya** [comment]
@lsaether could you please check if we have got access to merge queue?

**Comment by maltekliemann** [comment]
What's the blocker here? Also, can we get a notification when one of our PRs needs to resolve a merge conflict? https://docs.mergify.com/actions/comment/#request-for-action

**Comment by vivekvpandya** [comment]
> What's the blocker here? Also, can we get a notification when one of our PRs needs to resolve a merge conflict? https://docs.mergify.com/actions/comment/#request-for-action

We are waiting for merge queue access, however it seems like only admin can check that.

File: .mergify.yml
Comment (sea212) [inline .mergify.yml:5]:
```suggestion
      - "#approved-reviews-by>=1"
```

**Comment by sea212** [comment]
Mergify is installed in this repo: 
image

**Review by sea212 — APPROVED** [review]

**Pull request merged**
